process.py

```
@@ -0,0 +1,8 @@
+def process_data(data):
+    """
+    Applies a transformation to each item in the dataset.
+    """
+    result = []
+    for item in data:
+        result.append(complex_transformation(item))  # Apply transformation
```

Ruff (0.8.2)

Undefined name `complex_transformation`

(F821)